File: TheraLang.DAL/Entities/Resource.cs
using System.Collections.Generic;

namespace TheraLang.DAL.Entities
{
    public class Resource : BaseEntity
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public string Url { get; set; }
        
        public int CategoryId { get; set; }
        
        public virtual User User { get; set; }

        public virtual ResourceCategory ResourceCategory { get; set; }

        public virtual ICollection<ResourceProject> ResourceProjects { get; set; }
    }
}

File: TheraLang.Web/ClientApp/src/app/shared/models/resource/resource.ts
import { ResourceProjects } from "./resource-projects";
import { ResourceCategory } from "./resource-category";
import { User } from "../user/user";

export interface Resource {
  user: User;
  name: string;
  description: string;
  url?: string;
  file?: any;
  categoryId: number;
  authorName?: string;
  resourceCategory?: ResourceCategory;
  resourceProjects?: ResourceProjects;
  id: number;
  createdById: string;
  updatedById: string;
  createdDateUtc: Date;
  updatedDateUtc?: Date;
}
